Feature/login page frontend

ryanwooa opened the pull request:
Login page and Signup Page

ryanwooa committed:
Initialize Login and Signup Page

ryanwooa committed:
update

ryanwooa committed:
update content in the log&Sign component

ryanwooa committed:
update show password button

ryanwooa committed:
Update LogAndSign.jsx

ryanwooa committed:
Update index.css

ryanwooa committed:
Delete signup_pic2.svg

ryanwooa committed:
Update HomePage.jsx

ryanwooa committed:
update style

ryanwooa committed:
Adjust nav and CSS

ryanwooa committed:
Update App.jsx

ryanwooa committed:
Merge remote-tracking branch 'origin/develop' into feature/LoginPage-Frontend

ryanwooa committed:
Delete SignUpPage.jsx

ryanwooa committed:
Update App.jsx

tpan501 commented:
![image](https://github.com/UOA-CS732-SE750-Students-2024/project-group-noble-narwhals/assets/126734959/d07791e9-2025-4771-a55a-8b608d25d2cc)
works fine when the browser window size is below 100% but shows grey div when the size is >= 100%...
Maybe it's my browser problem

tpan501 commented:
![image](https://github.com/UOA-CS732-SE750-Students-2024/project-group-noble-narwhals/assets/126734959/03519467-16c5-4884-9222-9883093dbec2)
do we need to have a same gap between these three elements?

ryanwooa committed:
Edit path of image & update Navbar

- Absolute path for img
- Add div behind the navbar

ryanwooa committed:
Update size of CSS

ryanwooa committed:
Update size of gap

ryanwooa commented:
> ![image](https://private-user-images.githubusercontent.com/126734959/317244694-03519467-16c5-4884-9222-9883093dbec2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE1NzMxMTYsIm5iZiI6MTcxMTU3MjgxNiwicGF0aCI6Ii8xMjY3MzQ5NTkvMzE3MjQ0Njk0LTAzNTE5NDY3LTE2YzUtNDg4NC05MjIyLTk4ODMwOTNkYmVjMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyN1QyMDUzMzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMDRiOTRhZDM2ZGU5NzI4OWJlMjk2MjE2YmQxNjQ0YjkyNzA1NzViNjQwNWM3MTgxMDVhYTkyNmIxMjEwYWRhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.uLacyajARzUWwNt7nVtbQqQcU-sVFNcOOTmAzG-MG8s) do we need to have a same gap between these three elements?

Edit

ryanwooa commented:
> ![image](https://private-user-images.githubusercontent.com/126734959/317243874-d07791e9-2025-4771-a55a-8b608d25d2cc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE1NzMxMTYsIm5iZiI6MTcxMTU3MjgxNiwicGF0aCI6Ii8xMjY3MzQ5NTkvMzE3MjQzODc0LWQwNzc5MWU5LTIwMjUtNDc3MS1hNTVhLThiNjA4ZDI1ZDJjYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyN1QyMDUzMzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMjE5NjVjZTAzZGJlODgxMThiZThiZTMyZGFkNjcwYjkwZGJjMmVlMTljMTIxZWIwOTQ3MTIxMzRlNzUxNWFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.IHu6iBqvevACH44RkRy2MH4tJPURQoWWjJLy21IfJRc) works fine when the browser window size is below 100% but shows grey div when the size is >= 100%... Maybe it's my browser problem

>= 100%?
Now, the minimum size of page is 750 * 630, if the window size less than this number, the scrollbar will show.

ryanwooa committed:
Adjust Route for Userpage

- Home need max-width and User dont
- Restructure Route and Layout page

tpan501 commented:
Yeah now the login page looks much normal to me

How to reply other's response in this guthub page, not just comment..

zli213 commented on frontend/public/login_pic.png:
I remember you mentioned before about putting all photos under the assets folder, or did I remember it wrong? Also, I suggest creating a new folder named "image" to store all photos collectively.

ryanwooa replied:
When writing the code, I found out that it's better to put it in public. I will change it in next commit.

Then it's still possible to create an image folder in public to manage, and then create another folder for avatar

jlmjkfd reviewed (commented):
Does it look better if I set the buttons and the input the same size?
<div style="display:flex">
<img src="https://github.com/UOA-CS732-SE750-Students-2024/project-group-noble-narwhals/assets/6456953/3caff6ee-ad4d-4d26-ad55-96eb736a443a" width="200" />

<img src="https://github.com/UOA-CS732-SE750-Students-2024/project-group-noble-narwhals/assets/6456953/ea7d01c6-6ddf-4592-add8-2589393905a1" width="200" />
</div>

ryanwooa committed:
Change image path and update Readme & CSS adjust

- Change image path to /public/image
- Update readme about image path
- adjust width of login button

ryanwooa commented:
> Does it look better if I set the buttons and the input the same size?
> 
> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/6456953/317552684-3caff6ee-ad4d-4d26-ad55-96eb736a443a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2MjEzMTAsIm5iZiI6MTcxMTYyMTAxMCwicGF0aCI6Ii82NDU2OTUzLzMxNzU1MjY4NC0zY2FmZjZlZS1hZDRkLTRkMjYtYWQ1NS05NmViNzM2YTQ0M2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTAxNjUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGU5NzZhYTcyNDE4MzBhNjdmYTA4MzlkMmQzYzUzMjFiZTA3YjNkYTVmNWVlNjM0NGQ4ZTk4YWRhMmQwNjFjNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.yK4-1NGTyquU6h0OZ8cFh2oqWXtGq0GRtpD5cJcnEik"> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/6456953/317552607-ea7d01c6-6ddf-4592-add8-2589393905a1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2MjEzMTAsIm5iZiI6MTcxMTYyMTAxMCwicGF0aCI6Ii82NDU2OTUzLzMxNzU1MjYwNy1lYTdkMDFjNi02ZGRmLTQ1OTItYWRkOC0yNTg5MzkzOTA1YTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTAxNjUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OThhMDIyMGE5YmQwOTRmYjgzY2Y2NWJkZmY3NWVjZDk5MDBlN2ExYWJkOGM1ZjJmMzE1NDZjNTlmMDIxZTNmYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.DJxSXf6aXaPFb_Eeea7oiktOmEUj5uGN4hxK5MEAnsU">

Yes, the original one is too small. But the same width is a little bit wired? So I adjust it to a bit longer. Is it OK?
![image](https://github.com/UOA-CS732-SE750-Students-2024/project-group-noble-narwhals/assets/119087910/98cfe3c1-e621-45d7-9e76-60742cd4ed07)

ryanwooa committed:
Update Navbar.jsx

jlmjkfd commented:
> > Does it look better if I set the buttons and the input the same size?
> > <img alt="" width="200" src="https://private-user-images.githubusercontent.com/6456953/317552684-3caff6ee-ad4d-4d26-ad55-96eb736a443a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2MjEzMTAsIm5iZiI6MTcxMTYyMTAxMCwicGF0aCI6Ii82NDU2OTUzLzMxNzU1MjY4NC0zY2FmZjZlZS1hZDRkLTRkMjYtYWQ1NS05NmViNzM2YTQ0M2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTAxNjUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGU5NzZhYTcyNDE4MzBhNjdmYTA4MzlkMmQzYzUzMjFiZTA3YjNkYTVmNWVlNjM0NGQ4ZTk4YWRhMmQwNjFjNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.yK4-1NGTyquU6h0OZ8cFh2oqWXtGq0GRtpD5cJcnEik"> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/6456953/317552607-ea7d01c6-6ddf-4592-add8-2589393905a1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2MjEzMTAsIm5iZiI6MTcxMTYyMTAxMCwicGF0aCI6Ii82NDU2OTUzLzMxNzU1MjYwNy1lYTdkMDFjNi02ZGRmLTQ1OTItYWRkOC0yNTg5MzkzOTA1YTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTAxNjUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OThhMDIyMGE5YmQwOTRmYjgzY2Y2NWJkZmY3NWVjZDk5MDBlN2ExYWJkOGM1ZjJmMzE1NDZjNTlmMDIxZTNmYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.DJxSXf6aXaPFb_Eeea7oiktOmEUj5uGN4hxK5MEAnsU">
> 
> Yes, the original one is too small. But the same width is a little bit wired? So I adjust it to a bit longer. Is it OK? ![image](https://private-user-images.githubusercontent.com/119087910/317690586-98cfe3c1-e621-45d7-9e76-60742cd4ed07.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMwOTg1MDksIm5iZiI6MTcxMzA5ODIwOSwicGF0aCI6Ii8xMTkwODc5MTAvMzE3NjkwNTg2LTk4Y2ZlM2MxLWU2MjEtNDVkNy05ZTc2LTYwNzQyY2Q0ZWQwNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNFQxMjM2NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNTQwMGVkNTI5ZWNiNjIyMTczOWI2ZTUzOGJmMTcyNTk1OWFkODJmOThiODEwNGRjOWFhZWRlZDZjNDE0Y2RiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4un_XpNv8rIYNo9iPQHhXo7bymFUon41rpQp5UNwYHM)

If you search 'login page' on Google, most designs have inputs and buttons of the same length. Only a few of the lengths are different. I think the same length is better.

ryanwooa commented:
> > > Does it look better if I set the buttons and the input the same size?
> > > <img alt="" width="200" src="https://private-user-images.githubusercontent.com/6456953/317552684-3caff6ee-ad4d-4d26-ad55-96eb736a443a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2MjEzMTAsIm5iZiI6MTcxMTYyMTAxMCwicGF0aCI6Ii82NDU2OTUzLzMxNzU1MjY4NC0zY2FmZjZlZS1hZDRkLTRkMjYtYWQ1NS05NmViNzM2YTQ0M2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTAxNjUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGU5NzZhYTcyNDE4MzBhNjdmYTA4MzlkMmQzYzUzMjFiZTA3YjNkYTVmNWVlNjM0NGQ4ZTk4YWRhMmQwNjFjNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.yK4-1NGTyquU6h0OZ8cFh2oqWXtGq0GRtpD5cJcnEik"> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/6456953/317552607-ea7d01c6-6ddf-4592-add8-2589393905a1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2MjEzMTAsIm5iZiI6MTcxMTYyMTAxMCwicGF0aCI6Ii82NDU2OTUzLzMxNzU1MjYwNy1lYTdkMDFjNi02ZGRmLTQ1OTItYWRkOC0yNTg5MzkzOTA1YTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTAxNjUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OThhMDIyMGE5YmQwOTRmYjgzY2Y2NWJkZmY3NWVjZDk5MDBlN2ExYWJkOGM1ZjJmMzE1NDZjNTlmMDIxZTNmYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.DJxSXf6aXaPFb_Eeea7oiktOmEUj5uGN4hxK5MEAnsU">
> > 
> > 
> > Yes, the original one is too small. But the same width is a little bit wired? So I adjust it to a bit longer. Is it OK? ![image](https://private-user-images.githubusercontent.com/119087910/317690586-98cfe3c1-e621-45d7-9e76-60742cd4ed07.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMwOTg1MDksIm5iZiI6MTcxMzA5ODIwOSwicGF0aCI6Ii8xMTkwODc5MTAvMzE3NjkwNTg2LTk4Y2ZlM2MxLWU2MjEtNDVkNy05ZTc2LTYwNzQyY2Q0ZWQwNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNFQxMjM2NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNTQwMGVkNTI5ZWNiNjIyMTczOWI2ZTUzOGJmMTcyNTk1OWFkODJmOThiODEwNGRjOWFhZWRlZDZjNDE0Y2RiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4un_XpNv8rIYNo9iPQHhXo7bymFUon41rpQp5UNwYHM)
> 
> If you search 'login page' on Google, most designs have inputs and buttons of the same length. Only a few of the lengths are different. I think the same length is better.

OK, I'll adjust it to a same length

ryanwooa committed:
Update LogAndSign.jsx

Adjust the width of login button

ryanwooa committed:
Update LogAndSign.jsx
